node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -297,12 +298,13 @@ case class DataMessageHandler(
         handleInventoryMsg(invMsg = invMsg, peerMsgSender = peerMsgSender)
     }
 
-    resultF.failed.foreach {
-      case err =>
-        logger.error(s"Failed to handle data payload=${payload}", err)
+    resultF.failed.foreach { err =>
+      logger.error(s"Failed to handle data payload=${payload}", err)
     }
 
-    resultF
+    resultF.recoverWith {
+      case NonFatal(_) => Future.successful(this)
```

Comment:
LGTM, but let's add a unit test so we don't regress in the future?

Reply:
I think we need #2436 before we can do that

Reply:
I need this for the exact reason you mention on this PR, but I would still like a unit test for this after #2436 is in. I filed an issue #2501